shared/js/ui/base/model.es6.js

```diff
@@ -57,6 +57,9 @@ BaseModel.prototype = $.extend({},
       const lastValue = this[attr] || null
       this[attr] = val
 
+      // deep clone val in case it's something passed by reference
+      val = JSON.parse(JSON.stringify(val))
```

Comment:
I was considering doing this to `lastValue` as well. Not sure if it's worth the extra processing time for it, through.

Reply:
I think it's fine, lastValue seems something we wouldn't modify after sending a notification

Reply:
@andrey-p would you mind checking whether JSON.stringify(val) is defined though? If val is null, that method returns undefined, and JSON.parse() will throw a SyntaxError.

Reply:
Good find, lemme fix!